docs/source/jit.rst

```diff
@@ -493,9 +496,10 @@ Return
     ``return a, b``
 
     .. note::
-        there must be a return statement as the last member of the function
-        and return statements cannot appear anywhere else in the function. This
-        restriction will be removed in the future.
+        TorchScript allows returns in the following circumstances:
+           1. At the end of a function
+           2. In an if-statement where <true> and <false> both return
```

Comment:
It might be worth specifying that we don't allow early returns in the body of loops

Reply:
Well, it's probably easier to specify where it *is* allowed, since there are lots of places where it's not.

Reply:
Reading this it's not clear to me that something like

while cond:
     if early return:
         return hi

wouldn't work